feat(backend): Add database connection config and refactor logging and config management

gomezbc opened the pull request:
## Description
This PR adds the setup and methods to connect to the database. 

I had to move the code that reads configuration and logging settings to other files so that they can be used throughout the entire project.

I added the required `psycopg2-binary` package to connect to PostgreSQL with `sqlmodel`.

Modifies #156 #157 #158 database configuration to use the database connection from the configuration.yml file. It also removes the RepositoryManagerFactory singleton because it was re-using sessions across requests and that goes against what SqlAlchemy recommends:

> "Make sure you have a clear notion of where transactions begin and end, and keep transactions short, meaning, they end at the series of a sequence of operations, instead of being held open indefinitely." 
> https://docs.sqlalchemy.org/en/20/orm/session_basics.html#when-do-i-construct-a-session-when-do-i-commit-it-and-when-do-i-close-it

>The [Session](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session) is a mutable, stateful object that represents a single database transaction. An instance of [Session](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session) therefore cannot be shared among concurrent threads or asyncio tasks without careful synchronization. The [Session](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session) is intended to be used in a non-concurrent fashion, that is, a particular instance of [Session](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session) should be used in only one thread or task at a time.
> https://docs.sqlalchemy.org/en/20/orm/session_basics.html#is-the-session-thread-safe-is-asyncsession-safe-to-share-in-concurrent-tasks

<!-- Please tag the related issue `Fixes or Updates #issue_number`, if applicable. -->

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

gomezbc commented:
@CDiezRodriguez please run IP Check on `psycopg2-binary`

CDiezRodriguez reviewed (commented):
IP-Check: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/20738

gomezbc commented:
Feedback: Move database connection to config.yaml and create a configmap in helm chart

CDiezRodriguez commented on ichub-backend/requirements.txt:
```diff
@@ -27,6 +27,7 @@ MarkupSafe==2.0.1
 mdurl==0.1.2
 packaging==24.1
 pluggy==1.5.0
+psycopg2-binary==2.9.10
```
Why is it not included in the DEPENDENCIES file?

gomezbc replied:
It seems that the dash tool [url](https://repo.eclipse.org/service/local/artifact/maven/redirect?r=dash-licenses&g=org.eclipse.dash&a=org.eclipse.dash.licenses&v=LATEST) has been down since yesterday, and it wasn't able to detect the issue.
![image](https://github.com/user-attachments/assets/a0cda5e2-8287-4a5f-98bf-dc237f23dc69)

I will update the dependencies file locally and push it.

CDiezRodriguez reviewed (commented):
Awesome work, @gomezbc! I'm just wondering, why didn’t the workflow throw an error about the dependencies?

gomezbc commented:
Waiting for #156 #157 #158 to the be merged

gomezbc commented:
@CDiezRodriguez the PR is ready for review!

@sm29105 I’ve refactored our session management—details are in the PR description—but here’s the gist: I’ve removed the singleton session and stopped creating one session per class. Instead, each call to RepositoryManagerFactory.create() now spins up a fresh Session(engine) wrapped by RepositoryManager, ensuring a clean, isolated session per operation. In practice, you’ll now acquire a session simply by doing:
```py
with RepositoryManagerFactory.create() as repo:
```
instead of:
```py
with self.repositories as repos:
```
